asllib/aslspec/spec.ml

```diff
@@ -511,13 +511,27 @@ module ResolveRules = struct
   (** [lhs_of_conclusion] returns an expression representing the LHS of the
       conclusion judgment for relation definition with the given [name] and
       [input] arguments. This function assumes
-      [relation_names_arguments_if_exists_rule] has been called on the AST to
-      ensure that all input arguments have names. *)
+      [relation_named_arguments_if_exists_rule] has been called. *)
   let lhs_of_conclusion { Relation.name; input } =
-    let input_vars =
-      List.map (fun (name_opt, _) -> Var (Option.get name_opt)) input
+    let rec arg_of (name_opt, arg_type) =
+      if Option.is_some name_opt then Var (Option.get name_opt)
+      else
+        match arg_type with
+        | LabelledTuple { components } ->
+            let args = List.map arg_of components in
+            make_tuple args
+        | LabelledRecord { label_opt; fields } ->
+            let field_exprs =
+              List.map
+                (fun { name_and_type = field_name, field_type; _ } ->
+                  (field_name, arg_of (None, field_type)))
```

Comment:
I'm slightly unsure about the recursion here - if you have a field name, why does the type need `arg_of` to assign a name?